### PR TITLE
New user portal

### PR DESCRIPTION
**Pull request by ironBatista**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description

This new feature pretends to remove the old User portal with the new one using a similar stack as in web/client.

### CODE REVIEW

**Comment by ironArt3mis** [comment]
Rebase completed.

File: web/user/src/store/clientSession/aboutMe.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { action, Action, Thunk, thunk } from 'easy-peasy';
```
Comment (mmadariaga) [inline web/user/src/store/clientSession/aboutMe.ts:1]: I don't think we need this file

File: web/user/src/store/clientSession/recordLocutionService.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import { action, Action, Computed, computed, Thunk, thunk } from 'easy-peasy';
```
Comment (mmadariaga) [inline web/user/src/store/clientSession/recordLocutionService.ts:1]: I don't think we need this file

File: web/user/src/translations/es.json [hunk 3]
```diff
@@ -0,0 +1,1054 @@
+{
+    "%s emulated": "%s emulada",
```
Comment (mmadariaga) [inline web/user/src/translations/es.json:2]: This is inherited from klear, user portal was not based on it so, replace translations

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/portal/user/src/entities/CallForwardSetting/CallForwardSetting.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+const CallForwardSetting: EntityInterface = {
+  ...defaultEntityBehavior,
+  icon: AccountTreeIcon,
+  iden: 'CallForwardSetting',
+  title: _('Call forward setting', { count: 2 }),
+  path: '/my/call_forward_settings',
+  toStr: (row: any) => row.id,
+  properties,
+  columns,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/CallForwardSetting/CallForwardSetting.tsx:81]: `selectOptions,`
Wrapping selectOptions was required some time ago by ivoz-ui but it's not anymore

File: web/portal/user/.hygen.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const helpers = require("@irontec/ivoz-ui/hygen/lib");
+
+module.exports = {
+  templates: `${__dirname}/node_modules/@irontec/ivoz-ui/hygen/templates`,
+  helpers: {
+    ...helpers,
+    url: () => "https://10.60.75.33/api/user/docs.json",
```
Comment (Kaian) [inline web/portal/user/.hygen.js:7]: There is a private ip address here. Is this file required? generated? Does it work with localhost?

File: web/portal/user/src/entities/CallForwardSetting/SelectOptions.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  callback,
+  cancelToken,
+}): Promise<unknown> => {
+  return defaultEntityBehavior.fetchFks(
+    CallForwardSetting.path,
+    ['id'],
+    (data: any) => {
+      const options: any = {};
+      for (const item of data) {
+        options[item.id] = item.id;
```
Comment (mmadariaga) [inline web/portal/user/src/entities/CallForwardSetting/SelectOptions.ts:15]: Either it is not being used or you are displaying the id in the select boxes

File: web/portal/user/src/entities/Country/Country.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  toStr: (row: any) => {
+    const language = getI18n().language.substring(0, 2);
+    return row.name[language];
+  },
+  properties,
+  acl: {
+    ...defaultEntityBehavior.acl,
+    iden: 'Countries',
+  },
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/Country/Country.tsx:30]: Unwrap

File: web/portal/user/src/entities/Extension/Extension.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  icon: AccountTreeIcon,
+  iden: 'Extension',
+  title: _('Extension', { count: 2 }),
+  path: '/my/company_extensions',
+  acl: {
+    ...defaultEntityBehavior.acl,
+    iden: 'Extension',
+  },
+  toStr: (row: any) => row.name,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/Extension/Extension.tsx:18]: Unwrap

File: web/portal/user/src/entities/Preferences/ForeignKeyGetter.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  match,
+}): Promise<any> => {
+  const response: UserPropertyList<unknown> = {};
+
+  const promises = autoSelectOptions({
+    entities,
+    entityService,
+    cancelToken,
+    response,
+    skip: [],
```
Comment (mmadariaga) [inline web/portal/user/src/entities/Preferences/ForeignKeyGetter.tsx:19]: `skip: ['timezone']` or remove TimezoneSelectOptions call below
It think you're doing the request twice here.

File: web/portal/user/src/entities/Timezone/Timezone.tsx [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  icon: AccountTreeIcon,
+  iden: 'Timezone',
+  title: _('Timezone', { count: 2 }),
+  path: '/timezones',
+  acl: {
+    ...defaultEntityBehavior.acl,
+    iden: 'Timezones',
+  },
+  toStr: (row: any) => row.name,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/Timezone/Timezone.tsx:18]: Unwrap

File: web/portal/user/src/entities/User/User.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  ...defaultEntityBehavior,
+  icon: AccountTreeIcon,
+  iden: 'User',
+  title: _('Profile', { count: 2 }),
+  path: '/my/profile',
+  localPath: '/profile',
+  toStr: (row: any) => row.id,
+  properties,
+  columns,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/User/User.tsx:80]: Unwrap

File: web/portal/user/src/entities/UsersCdr/UsersCdr.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+const UsersCdr: EntityInterface = {
+  ...defaultEntityBehavior,
+  icon: AccountTreeIcon,
+  iden: 'UsersCdr',
+  title: _('Calls', { count: 2 }),
+  path: '/my/call_history',
+  toStr: (row: any) => row.id,
+  properties,
+  columns,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/UsersCdr/UsersCdr.tsx:67]: Unwrap

File: web/portal/user/src/entities/VoiceMail/VoiceMail.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import AccountTreeIcon from '@mui/icons-material/AccountTree';
+import EntityInterface from '@irontec/ivoz-ui/entities/EntityInterface';
+import _ from '@irontec/ivoz-ui/services/translations/translate';
+import defaultEntityBehavior from '@irontec/ivoz-ui/entities/DefaultEntityBehavior';
+import selectOptions from './SelectOptions';
+
+const VoiceMail: EntityInterface = {
+  ...defaultEntityBehavior,
+  icon: AccountTreeIcon,
+  iden: 'Voice Mail',
```
Comment (mmadariaga) [inline web/portal/user/src/entities/VoiceMail/VoiceMail.tsx:10]: `iden: 'Voicemail',` 
This should match entity name in the API

File: web/portal/user/src/entities/VoiceMail/VoiceMail.tsx [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  title: _('Voice Mail', { count: 2 }),
+  path: '/my/company_voicemails',
+  acl: {
+    ...defaultEntityBehavior.acl,
+    iden: 'Voice Mail',
+  },
+  toStr: (row: any) => row.name,
+  selectOptions: (props, customProps) => {
```
Comment (mmadariaga) [inline web/portal/user/src/entities/VoiceMail/VoiceMail.tsx:18]: Unwrap

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/portal/user/.hygen.js (hunk 5 above)
Comment (ironBatista) [reply web/portal/user/.hygen.js:7]: That file is only used for development purposes, anyway, there is no need to track such a file in git I will change .gitignore to avoid it.

**Review by mmadariaga — APPROVED** [review]

**Comment by ironArt3mis** [comment]
Rebase completed.

**Comment by ironArt3mis** [comment]
Rebase completed.

**Review by Kaian — APPROVED** [review]
